src/dfx/src/actors/pocketic.rs

```diff
@@ -0,0 +1,323 @@
+        // If we have a port, send that we're already ready! Yeah!
+        if let Some(port) = self.port {
+            msg.0.do_send(PortReadySignal {
+                url: format!("http://localhost:{port}/instances/0/"),
+            });
+        }
+
+        self.ready_subscribers.push(msg.0);
+    }
+}
+
+impl Handler<signals::PocketIcRestarted> for PocketIc {
+    type Result = ();
+
+    fn handle(
+        &mut self,
+        msg: signals::PocketIcRestarted,
+        _ctx: &mut Self::Context,
+    ) -> Self::Result {
+        self.port = Some(msg.port);
+        self.send_ready_signal(msg.port);
+    }
+}
+
+impl Handler<Shutdown> for PocketIc {
+    type Result = ResponseActFuture<Self, Result<(), ()>>;
+
+    fn handle(&mut self, _msg: Shutdown, _ctx: &mut Self::Context) -> Self::Result {
+        // This is just the example for ResponseActFuture but stopping the context
+        Box::pin(
+            async {}
+                .into_actor(self) // converts future to ActorFuture
+                .map(|_, _act, ctx| {
+                    ctx.stop();
+                    Ok(())
+                }),
+        )
+    }
+}
+
+fn pocketic_start_thread(
+    logger: Logger,
+    config: Config,
+    addr: Addr<PocketIc>,
+    receiver: Receiver<()>,
+) -> DfxResult<std::thread::JoinHandle<()>> {
+    let thread_handler = move || {
+        // Start the process, then wait for the file.
+        let pocketic_path = config.pocketic_path.as_os_str();
+
+        // form the pocket-ic command here similar to the ic-starter command
+        let mut cmd = std::process::Command::new(pocketic_path);
+        if let Some(port) = config.port {
+            cmd.args(["--port", &port.to_string()]);
+        };
+        cmd.args([
+            "--port-file",
+            &config.port_file.to_string_lossy(),
+            "--ttl",
+            "2592000",
+        ]);
+        if !config.verbose {
+            cmd.env("RUST_LOG", "error");
+        }
+        cmd.stdout(std::process::Stdio::inherit());
+        cmd.stderr(std::process::Stdio::inherit());
+
+        loop {
+            let _ = std::fs::remove_file(&config.port_file);
+            let last_start = std::time::Instant::now();
+            debug!(logger, "Starting PocketIC...");
+            let mut child = cmd.spawn().expect("Could not start PocketIC.");
+
+            let port = PocketIc::wait_for_port_file(&config.port_file).unwrap();
+
+            if let Err(e) = block_on_initialize_pocketic(port, logger.clone()) {
+                error!(logger, "Failed to initialize replica: {:#}", e);
+                let _ = child.kill();
+                let _ = child.wait();
+                if receiver.try_recv().is_ok() {
+                    debug!(logger, "Got signal to stop.");
+                    break;
+                } else {
+                    continue;
+                }
+            }
+            addr.do_send(signals::PocketIcRestarted { port });
```

Comment:
When running without `--background`, `dfx start` indicates pretty clearly when it's ready:
```
$ dfx start --clean 
Running dfx start for version 0.20.1-beta.0+rev26.e12a2404
Using shared network 'local' defined in /Users/ericswanson/.config/dfx/networks.json
Initialized replica.
Dashboard: http://localhost:59729/_/dashboard
```

However, when running with PocketIC, it can give the idea that it got stuck:
```
$ dfx start --clean --pocketic
Running dfx start for version 0.20.1-beta.0+rev26.e12a2404
Using shared network 'local' defined in /Users/ericswanson/.config/dfx/networks.json
```

Can we display some indication that it's ready to go?